src/inspect.rs

```diff
@@ -13,6 +13,44 @@ use crate::cli::*;
 use crate::errors::CliError::*;
 use crate::input::*;
 
+fn handle_query(
+    query: &Rule,
+    query_all: bool,
+    all_params: &[Param],
+    authorizer: &mut Authorizer,
```

Comment:
Why authorizer is "mutable"? Querying should not change things (it is read-only), no?

Reply:
`query` does mutate the authorizer: it triggers datalog evaluation. running `query` on an authorizer that has already performed authorization won't change anything, but running it on a brand new authorizer will trigger fact generation. In addition to mutating the set of generated facts, `query` also modifies the timer that keep tracks of time spent generating facts, and the number of iterations of fact generation (those values are used to abort evaluation if it takes too much time / resources).

Here it's the type system keeping us honest about what actually happens, even though the behaviour is not totally obvious.